web/client/utils/CatalogUtils.js

```diff
@@ -79,7 +136,17 @@ const converters = {
                     const URI = isArray(dc.URI) ? dc.URI : (dc.URI && [dc.URI] || []);
                     let thumb = head([].filter.call(URI, (uri) => {return uri.name === "thumbnail"; }) ) || head([].filter.call(URI, (uri) => !uri.name && uri.protocol?.indexOf('image/') > -1));
                     thumbURL = thumb ? thumb.value : null;
-                    wms = head([].filter.call(URI, (uri) => { return uri.protocol && (uri.protocol.match(/^OGC:WMS-(.*)-http-get-map/g) || uri.protocol.match(/^OGC:WMS/g)); }));
+                    wms = head([].filter.call(URI, (uri) => {
+                        return uri.protocol && (
+                            /** wms protocol params are explicitly defined as attributes (INSPIRE)*/
+                            uri.protocol.match(/^OGC:WMS-(.*)-http-get-map/g) ||
+                            uri.protocol.match(/^OGC:WMS/g) ||
+                            /** wms protocol params must be extracted from the element text (RNDT / INSPIRE) */
+                            uri.protocol.match(/serviceType\/ogc\/wms/g));
+                    }));
+                }
+                if (wms && wms.protocol.match(/serviceType\/ogc\/wms/g) && wms.description === "access point") {
+                    wms = extractWMSParamsFromURL(wms);
```

Comment:
We prefer to not not add too many if/then that modify objects in a function, or with the evolution of the code we loose pieces and we can not undestand while reading the final role. This is a quick change that you should test. 
```suggestion
                            /** wms protocol params must be extracted from the element text (RNDT / INSPIRE) */
                            uri.protocol.match(/serviceType\/ogc\/wms/g) && extractWMSParamsFromURL(wms));
                    }));
                }
```